merkle_tree_wasm/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+#![allow(non_snake_case)]
+mod merkle_tree_wasm;
+
+extern crate console_error_panic_hook;
+
+use ark_ff::{BigInteger, PrimeField};
+use ark_serialize::CanonicalSerialize;
+use num_bigint::BigUint;
+use poseidon::constants::secp256k1_w3;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+pub use crate::merkle_tree_wasm::{MerkleProofBytes, MerkleTree};
+
+fn internal_generate_merkle_proof<F: PrimeField>(
+    leaves: Vec<String>,
+    leaf: String,
+    depth: usize,
+) -> MerkleProofBytes {
+    let leaves = leaves.clone();
+
+    let mut padded_leaves = leaves.clone();
+    // Pad the leaves to equal the size of the tree
+    padded_leaves.resize(1 << depth, "0".to_string());
+
+    const ARITY: usize = 2;
+    const WIDTH: usize = ARITY + 1;
+
+    let mut tree = MerkleTree::<F, WIDTH>::new(secp256k1_w3());
+
+    // Insert all the leaves into the tree
+    for leaf in &padded_leaves {
+        // Converting String to F
+        let leaf_hex = hex::decode(leaf.replace("0x", "")).unwrap();
+        let leaf_bytes = F::from(BigUint::from_bytes_be(&leaf_hex));
+        tree.insert(leaf_bytes);
+    }
+
+    tree.finish();
+
+    let leaf_hex = hex::decode(leaf.replace("0x", "")).unwrap();
+    let leaf_prime_field = F::from(BigUint::from_bytes_be(&leaf_hex));
+
+    let proof = tree.create_proof(leaf_prime_field);
+
+    // Encode the Merkle Proof output with BigEndian
+    let mut merkle_siblings = Vec::with_capacity(1 * depth);
+    let mut merkle_indices = Vec::with_capacity(1 * depth);
+    let siblings_bytes = proof
+        .siblings
+        .iter()
+        .flat_map(|sibling| sibling.into_bigint().to_bytes_be())
+        .collect::<Vec<u8>>();
+
+    let indicies_bytes = proof
+        .path_indices
+        .iter()
+        .map(|i| F::from(*i as u32).into_bigint().to_bytes_be())
+        .flatten()
```

Comment:
why not using `flat_map` again here like for `siblings_bytes`?
```suggestion
        .flat_map(|i| F::from(*i as u32).into_bigint().to_bytes_be())
```